app/src/main/java/com/kylehanish/todo/MainActivity.java

```diff
@@ -0,0 +1,237 @@
+    private iTodoRepository mRepository;
+    private LinearLayoutManager layoutManager;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        mUnbinder = ButterKnife.bind(this);
+
+        mContext = this;
+
+        setSupportActionBar(toolbar);
+
+        mRepository = new TodoSQLRepository(mContext);
+
+        SetBundleValues(savedInstanceState);
+        getListData();
+
+        mAdapter = new TodoItemArrayAdapter(this, R.layout.list_item_todo, mTodoItems, this);
+        recyclerView.setAdapter(mAdapter);
+        recyclerView.setLayoutManager(new LinearLayoutManager(this));
+//        recyclerView.setOnLongClickListener(new View.OnLongClickListener() {
+//            @Override
+//            public boolean onLongClick(View v) {
+//                if (getFragmentManager().findFragmentByTag(PreviewDialogFragment.TAG) == null) {
+//                    PreviewDialogFragment dialogFragment = new PreviewDialogFragment();
+//
+//                    int position = recyclerView.getChildLayoutPosition(v);
+//                    TodoItem item = mTodoItems.get(position);
+//
+//                    if (item != null) {
+//                        Bundle args = new Bundle();
+//                        args.putParcelable(PreviewDialogFragment.BUNDLE_TODO_ITEM, item);
+//                        dialogFragment.setArguments(args);
+//                    }
+//
+//                    dialogFragment.show(getFragmentManager(), PreviewDialogFragment.TAG);
+//                }
+//
+//                return true;
+//            }
+//        });
+    }
+
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+
+        if (id == R.id.add_new_item) {
+            showEditorDialog(null);
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putParcelableArrayList(BUNDLE_TODO_ITEMS, mTodoItems);
+
+    }
+
+    private void SetBundleValues(Bundle args) {
+
+        if (args != null) {
+            if (args.containsKey(BUNDLE_TODO_ITEMS)) {
+                mTodoItems = args.getParcelableArrayList(BUNDLE_TODO_ITEMS);
+            }
+        }
+
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        getListData();
+        SetListVisibilty();
+    }
+
+
+    private void getListData() {
+        if (mTodoItems == null) {
+            mTodoItems = mRepository.getTodoItems(null, null, null, null, null);
+        }
+    }
+
+    private void SetListVisibilty() {
+        if (mTodoItems != null && mTodoItems.size() > 0) {
```

Comment:
This nested if could probably be cleaned up a little. Not sure if you need to check the current visibility before setting it?

Reply:
I agree, it could be cleaned up. I also agree we could go without the visibility check which would help clean up code